### PR TITLE
Implement RRM setup flow

### PR DESCRIPTION
**Pull request by nfmohit**
## Summary

Addresses issue:

- #8800 

### Unrelated changes

This PR includes the following unrelated changes:
- Updated `PublicationCreate` component to set the created publication in store.
- Removed dispatch to `syncPublicationOnboardingState` in `getPublication` resolver, as suggested [here](https://github.com/google/site-kit-wp/pull/9017#discussion_r1693356104).
- Removed enabled feature flag as a requirement for module registration. This isn't what we usually do for module registration. The feature flag guard on the server side should automatically prevent the module from appearing.

### Deviations from IB
- Added a `selectPublication` action that sets publication in store.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/reader-revenue-manager/components/common/PublicationCreate.js [hunk 1]
```diff
@@ -43,21 +39,27 @@ export default function PublicationCreate( { onCompleteSetup } ) {
...
 	const hasPublication = publications && publications.length > 0;
 
-	const handleCompleteSetupClick = () => {
-		// TODO: Track event for completing setup.
+	const handleCompleteSetupClick = useCallback( async () => {
+		if ( ! hasPublication ) {
+			return;
+		}
+
+		await selectPublication( publications[ 0 ] );
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/common/PublicationCreate.js:55]: The change in this component ensures that the created publication is selected in store before saving settings.

File: assets/js/modules/reader-revenue-manager/datastore/publications.js [hunk 2]
```diff
@@ -260,7 +307,6 @@ const baseResolvers = {
 			.getPublications();
 		if ( publications === undefined ) {
 			yield fetchGetPublicationsStore.actions.fetchGetPublications();
-			yield baseActions.syncPublicationOnboardingState();
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/datastore/publications.js:263]: This has been removed based on the discussion [here](https://github.com/google/site-kit-wp/pull/9017#discussion_r1693356104).

File: assets/js/modules/reader-revenue-manager/components/setup/SetupMain.stories.js [hunk 3]
```diff
@@ -19,29 +19,99 @@
...
+NoPublications.storyName = 'No publications';
+NoPublications.args = {
+	setupRegistry: ( registry ) => {
+		registry
+			.dispatch( MODULES_READER_REVENUE_MANAGER )
+			.receiveGetPublications( [] );
+	},
+};
+NoPublications.scenario = {
+	// No parameters are needed for this scenario, but we still need to define the scenario object to include the story in the VRT suite.
```
Comment (kuasha420) [inline assets/js/modules/reader-revenue-manager/components/setup/SetupMain.stories.js:51]: I don't feel like this comment is needed. As time goes on, we should have the collective knowledge that the parameter doesn't need that label or anything. While this is useful comment now, it's also kind of repetitive and bloat-y.

File: assets/js/modules/reader-revenue-manager/index.js [hunk 4]
```diff
@@ -33,20 +33,11 @@ import DashboardMainEffectComponent from './components/DashboardMainEffectCompon
...
 
 export { registerStore } from './datastore';
 
-const isRrmModuleEnabled =
-	( func ) =>
-	( ...args ) => {
-		if ( isFeatureEnabled( 'rrmModule' ) ) {
-			func( ...args );
-		}
-	};
```
Comment (kuasha420) [inline assets/js/modules/reader-revenue-manager/index.js:47]: The explained rationale for this removal in the PR description SGTM :+1:

**Review by kuasha420 — CHANGES_REQUESTED** [review]
Excellent work @nfmohit ! This is good to go, I have a couple of nitpicks, and they're super minor so I'll push a commit to address those and get it merged to unblock other tickets. Cheers.

**Review by kuasha420 — APPROVED** [review]
Nice work @nfmohit. LGTM :tada:

**Pull request merged**